core-api/src/main/kotlin/com/jakemoore/datakache/api/result/handler/RejectableUpdateGenericDocResultHandler.kt

```
@@ -0,0 +1,64 @@
+            // METRICS
+            DataKacheMetrics.getReceiversInternal().forEach(MetricsReceiver::onDocRejectableUpdateNotFoundFail)
+
+            // In the middle of this update we discovered that the document does not exist.
+            // (it was likely deleted by another thread or task before this operation could complete)
+            return Failure(
+                ResultExceptionWrapper(
+                    message = "Update operation failed: Generic Document not found",
+                    exception = e,
+                )
+            )
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)